src/Build/Xml/XmlReaderExtension.cs

```diff
@@ -71,15 +83,61 @@ public void Dispose()
             _stream?.Dispose();
         }
 
+        /// <summary>
+        /// Returns <see cref="PropertyInfo"/> of the "Normalization" internal property on the given <see cref="XmlReader"/>-derived type.
+        /// </summary>
+        private static PropertyInfo GetNormalizationPropertyInfo(Type xmlReaderType)
+        {
+            PropertyInfo propertyInfo = _normalizationPropertyInfo;
+            if (propertyInfo == null)
+            {
+                BindingFlags bindingFlags = BindingFlags.NonPublic | BindingFlags.SetProperty | BindingFlags.Instance;
+                propertyInfo = xmlReaderType.GetProperty("Normalization", bindingFlags);
```

Comment:
Nit: why do you need to introduce a propertyInfo local variable as opposed to just modifying _normalizationPropertyInfo as relevant?

Reply:
It's a volatile field so I want to access it only when I really need to to avoid unnecessary memory barriers.

```C#
if (_normalizationPropertyInfo == null)
{
  ..
}
return _normalizationPropertyInfo;
```

would read the field twice on the fast path. The cost is small, limited to still rather exotic platforms, and negligible in the grand scheme of things, but a good practice anyway.

(In retrospect, explicit `Volatile.Read` and `Volatile.Write` may have been a better choice for readability.)